src/main.js

```diff
@@ -24,31 +23,22 @@ window.onload = randomBookGenerator;
 randomButton.addEventListener('click', randomBookGenerator);
 
 // Create your event handlers and other functions here 👇
-function randomTitle(){
-  var titleIndex = getRandomIndex(titles);
-  coverTitle.innerText = titles[titleIndex];
-}
 
-function randomCoverImage(){
+
+function randomBookGenerator() {
   var coverIndex = getRandomIndex(covers);
   coverImage.src = covers[coverIndex];
-}
 
-function randomDescriptor1(){
+  var titleIndex = getRandomIndex(titles);
+  coverTitle.innerText = titles[titleIndex];
+
   var tag1Index = getRandomIndex(descriptors);
   tagline1.innerText = descriptors[tag1Index];
-}
 
-function randomDescriptor2(){
   var tag2Index = getRandomIndex(descriptors);
   tagline2.innerText = descriptors[tag2Index];
-}
 
-function randomBookGenerator() {
-  randomTitle();
-  randomCoverImage();
-  randomDescriptor1();
-  randomDescriptor2();
+  currentCover = new Cover(covers[coverIndex], titles[titleIndex], descriptors[tag1Index], descriptors[tag2Index])
```

Comment:
Refactored so that the randomBookGenerator will reassign the values of the currentCover class instance, depending on the indexes that the function generates.